This combines createListing and updateListing

fulldecent opened the pull request:
Fixes #17, fixes #18

Notes that affect the application:

1. Previously, the contract had an undocumented feature where existing listings could not be updated to increase the price. This PR removes that limitation. (As noted in documentation, this does not create a front-running risk because the buyer will always specify a price.)
2. The `updateListing` function is removed, just use `createListing` wherever you would have used `updateListing`.
3. The `ItemUpdated` event is removed. Instead, you will now see `ItemListed` events. If you had previously been relying on the count of `ItemListed` events for some reason and you believe this change hurts you, please let's discuss further.

ghoul-sol commented:
Code of two functions could be combined into one internal function. I think it will be better to keep create/update interface and events so subgraph and UI do not need to change and we can release marketplace faster.

fulldecent commented:
@ghoul-sol Got it. Rebased and updated PR to maintain two separate external function but refactor to merge duplicate functionality into one internal function utilized by both.
